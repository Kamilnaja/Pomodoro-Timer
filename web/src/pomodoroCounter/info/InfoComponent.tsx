import { ConnectedTodayStatsContainer } from '../../stats/todayStats/containers/TodayStatsContainer';
import { CounterState } from '../store/enums/CounterState';
import './info.scss';
import { InfoComponentProps } from './InfoComponentProps';

export const InfoComponent = (props: InfoComponentProps) => {
  let info = '';
  switch (props.currentState) {
    case CounterState.END:
      info = 'Select new pomodoro or break from blue buttons and start timer';
      break;
    case CounterState.RUNNING:
      info = 'Timer in progress';
      break;
    case CounterState.PAUSE:
      info = 'Timer paused';
      break;
  }
  return (
    <div className="info mt-4">
      {props.auth.isLoggedIn ? (
        <div className="text-white text-center">
          <span>{info}</span>
          <ConnectedTodayStatsContainer />
        </div>
      ) : (
        <div className="text-danger text-center">
          <span>Please login to save pomodoros</span>
        </div>
      )}
    </div>
  );
};
