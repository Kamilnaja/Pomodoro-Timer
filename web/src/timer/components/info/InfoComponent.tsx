import { TimerState } from '../../store/enums/timerEnum';
import './info.scss';

export const InfoComponent = (props: { currentState: TimerState }) => {
  let info = '';
  switch (props.currentState) {
    case TimerState.BREAK_END:
      info = 'Start new pomodoro';
      break;
    case TimerState.POMODORO_END:
      info = 'Start break';
      break;
    case TimerState.POMODORO_RUNNING:
      info = 'Hard work';
      break;
    case TimerState.BREAK_RUNNING:
      info = 'Do not work';
      break;
    case TimerState.POMODORO_PAUSE:
      info = 'Pomodoro paused';
      break;
    case TimerState.BREAK_PAUSE:
      info = 'Break pause';
      break;
  }
  return (
    <div className="info">
      <hr className="text-white" />
      <span className="text-white text-center">{info}</span>
    </div>
  );
};
