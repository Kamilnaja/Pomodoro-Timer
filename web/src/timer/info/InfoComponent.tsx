import TodayStatsContainer from '../../stats/todayStats/containers/TodayStatsContainer';
import { TimerState } from '../state/enums/timerEnum';
import './info.scss';
import { InfoComponentProps } from './InfoComponentProps';

export const InfoComponent = (props: InfoComponentProps) => {
  let info = '';
  switch (props.currentState) {
    case TimerState.BREAK_END:
      info = 'Start new pomodoro';
      break;
    case TimerState.POMODORO_END:
      info = 'Start break';
      break;
    case TimerState.POMODORO_RUNNING:
      info = 'Hard work';
      break;
    case TimerState.BREAK_RUNNING:
      info = 'Do not work';
      break;
    case TimerState.POMODORO_PAUSE:
      info = 'Pomodoro paused';
      break;
    case TimerState.BREAK_PAUSE:
      info = 'Break pause';
      break;
  }
  return (
    <div className="info">
      {props.authState.isLoggedIn ? (
        <div className="text-white text-center">
          <span>{info}</span>
          <TodayStatsContainer />
        </div>
      ) : (
        <div className="text-danger text-center">
          <span>Please login to save pomodoros</span>
        </div>
      )}
    </div>
  );
};
