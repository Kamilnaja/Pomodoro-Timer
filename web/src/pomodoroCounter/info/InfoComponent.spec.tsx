import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { createAuth, createCurrentState } from '../testing/pomodoroCounter.test.data';
import { InfoComponent } from './InfoComponent';

describe('InfoComponent', () => {
  const infoComponent = shallow(<InfoComponent currentState={createCurrentState()} auth={createAuth()} />);

  it('Should match snapshot', () => {
    expect(shallowToJson(infoComponent)).toMatchSnapshot();
  });
});
