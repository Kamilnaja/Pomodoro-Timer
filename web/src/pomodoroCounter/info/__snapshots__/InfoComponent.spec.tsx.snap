// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoComponent Should match snapshot 1`] = `
<div
  className="info mt-4"
>
  <div
    className="text-danger text-center"
  >
    <span>
      Please login to save pomodoros
    </span>
  </div>
</div>
`;
